src/uct/api/uct.h

```diff
@@ -2061,6 +2062,49 @@ ucs_status_t uct_ep_connect_to_ep(uct_ep_h ep, const uct_device_addr_t *dev_addr
 ucs_status_t uct_md_query(uct_md_h md, uct_md_attr_t *md_attr);
 
 
+/**
+ * @ingroup UCT_MD
+ * @brief UCT allocation parameter specification field
+ *
+ * The enumeration allows specifying which fields in @ref uct_mem_alloc_param_t
+ * are present.
+ */
+typedef enum {
+    UCT_ALLOC_ATTR_FIELD_FLAGS    = UCS_BIT(0), /**< Allocation flags field*/
+    UCT_ALLOC_ATTR_FIELD_MEM_TYPE = UCS_BIT(1)  /**< Memory type field */
+} uct_alloc_attr_t;
+
+
+/**
+ * @ingroup UCT_MD
+ * @brief Allocation characteristics parameter passed to @ref uct_mem_alloc,
+ *        @ref uct_md_mem_alloc
+ *
+ * The structure @ref uct_mem_alloc_param_t is used to specify the memory type
+ * of the allocation, and flags indicating other characteristics of the
+ * allocation.
+ */
+typedef struct {
+    /**
+     * Mask of valid fields in this structure, using bits from
+     * @ref uct_alloc_attr_t. Fields not specified in this mask will be
+     * ignored
+     */
+    uint32_t       alloc_attr_mask;
+
+    /**
+     * Memory allocation flags, see @ref uct_md_mem_flags
+     */
+    unsigned flags;
```

Comment:
1. align on column
2. 2. pls specify which field_mask flag is enabling it

Reply:
Done